### PR TITLE
Use prettier 3

### LINKED ISSUE #5822: Update prettier across the repo to Miniflare's version, run prettify, and add commit to git blame ignore revs

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Of special note: we delete a bunch of lines from `.prettierignore`, since Prettier 3 can read `.gitignore` files

Fixes #5822. 

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because:

### CODE REVIEW

File: templates/worker-d1/src/index.ts [hunk 1]
```diff
@@ -20,26 +20,28 @@ app.get('/', async c => {
...
+				<body style="padding: 1em 2em">
+					${tables.results.map(
+						row =>
+							html`<div>
+								<a href="${new URL(`/api/${row.name}`, c.req.url)}">${row.name}</a>
+							</div>`
+					)}
+				</body>
+			</html>`
+	);
```
Comment (petebacondarwin) [inline templates/worker-d1/src/index.ts:44]: That's interesting!

**Review by petebacondarwin — APPROVED** [review]

File: packages/miniflare/.eslintrc.js [hunk 2]
```diff
@@ -3,10 +3,8 @@ module.exports = {
 	extends: ["plugin:prettier/recommended"],
 	plugins: ["import", "es"],
 	rules: {
-		"import/order": ["warn", { alphabetize: { order: "asc" } }],
 		"no-undef-init": "off",
-		"sort-imports": ["warn", { ignoreDeclarationSort: true }],
```
[the remark this answers is not in the document]
Comment (penalosa) [reply packages/miniflare/.eslintrc.js:8]: Yeah, this is covered by the `@ianvs/prettier-plugin-sort-imports` plugin

File: packages/miniflare/src/index.ts [hunk 3]
```diff
@@ -1596,9 +1604,10 @@ export class Miniflare {
 		// on the Content-Encoding header rather than trying to infer it from the body.
 		// Technically, at this point, this a malformed response so let's remove the header
 		// Retain it as MF-Content-Encoding so we can tell the body was actually compressed.
-		const contentEncoding = response.headers.get('Content-Encoding');
-		if (contentEncoding) response.headers.set('MF-Content-Encoding', contentEncoding);
-		response.headers.delete('Content-Encoding');
+		const contentEncoding = response.headers.get("Content-Encoding");
+		if (contentEncoding)
+			response.headers.set("MF-Content-Encoding", contentEncoding);
```
[the remark this answers is not in the document]
Comment (penalosa) [reply packages/miniflare/src/index.ts:1609]: I'm tempted to turn on https://eslint.org/docs/latest/rules/curly—what do you think?

[the remark this answers is not in the document]
Comment (petebacondarwin) [reply packages/miniflare/src/index.ts:1609]: I would be in favour of this although I accept it is contentious. Let's discuss in a team meeting and do in a follow up PR if everyone agrees.

**Review by petebacondarwin — APPROVED** [review]

**Review by petebacondarwin — APPROVED** [review]

**Review by CarmenPopoviciu — APPROVED** [review]

**Pull request merged**